src/app/archive/archive_lib/processor.ml

```diff
@@ -1100,24 +1177,44 @@ module Block = struct
                           (module Conn)
                           fee_transfer `Via_coinbase
                       in
-                      Block_and_internal_command.add_with_balance
+                      let%bind receiver_id =
+                        Public_key.add_if_doesn't_exist
+                          (module Conn)
+                          coinbase.receiver
+                      in
+                      let%bind receiver_balance_id =
+                        Balance.add
```

Comment:
it's possible there's already a `balance` entry with the same receiver and balance amount, so adding a new entry will create an unneeded duplicate

Reply:
Ah right, I'll switch these all to the `add_if_doesn't_exist` pattern